docs/how-to-guides/block-tutorial/extending-the-query-loop-block.md

```diff
@@ -0,0 +1,195 @@
+-   `taxQuery` - Shows available taxonomies filters for the currently selected post type.
+-   `author` - Shows an input field to filter the query by author.
+-   `search` - Shows an input filed to filter the query by keywords.
+
+In our case, the property would look like this:
+
+```js
+{
+	/** ...variation properties */
+	allowedControls: [ 'inherit', 'order', 'taxQuery', 'search' ],
+}
+```
+
+Notice that we have also disabled the `postType` control: when the user selects our variation, why show them a confusing dropdown to change the post type? On top of that it might break the block as we can implement custom controls, as we'll see shortly.
+
+### Adding additional controls
+
+Because our plugin uses custom attributes that we need to query, we want the users of our block to be able to select those instead of the ones we have just disabled from the core inspector controls. We can do this via a [React HOC](https://reactjs.org/docs/higher-order-components.html), like so:
+
+```jsx
+export const withBookQueryControls = ( BlockEdit ) => ( props ) => {
+	// We only want to add these controls if it is our variation,
+	// so here we can implement a logic to check for that, similiar
+	// to the `isActive` function described above.
+	return isMyBooksVariation( props ) ? (
+		<>
+			<BlockEdit { ...props } />
+			<InspectorControls>
+				<BookAuthorSelector /> { /** Our custom component */ }
+			</InspectorControls>
+		</>
+	) : (
+		<BlockEdit { ...props } />
+	);
+};
+
+addFilter( 'editor.BlockEdit', 'core/query', withBookQueryControls );
+```
+
+Of course, you'll be responsible for implementing the logic of your control (you might want to take a look at [`@wordpress/components`](https://www.npmjs.com/package/@wordpress/components) to make your controls fit seamlessly within the Gutenberg UI). Any extra parameter you assign within the `query` object inside the blocks attributes can be used to create a custom query according to your needs, with a little extra effort.
+
+Currently, you'll likely have to implement slightly different paths to make the query behave correctly both on the front-end side (i.e. on the end user's side) and to show the correct preview on the editor side.
+
+```js
+{
+	/** ...variation properties */
+	attributes: {
+		/** ...variation attributes */
+		query: {
+			/** ...more query settings if needed */
+			postType: 'book',
+			/** Our custom query parameter */
+			bookAuthor: 'J. R. R. Tolkien'
+		}
+	}
+}
+```
+
+### Making your custom query work on the front-end side
+
+The Query Loop block functions mainly through the Post Template block which receives the attributes and builds the query from there. Other first-class children of the Query Loop block (such as the Pagination block) behave in the same way. They build their query and then expose the result via the filter [`query_loop_block_query_vars`](https://developer.wordpress.org/reference/hooks/query_loop_block_query_vars/).
```

Comment:
The link `https://developer.wordpress.org/reference/hooks/query_loop_block_query_vars` is currently broken. I assume that will only appear with the release of 6.1?

Reply:
I'm actually unsure of how it works, that's something I needed to clarify with you folks. I added the URL using the convention I saw, assuming that some sort of script reads the JSDoc of a hook and publishes the updated references. Would that be correct? Or who would I have to ask?